packages/grid/src/vaadin-grid-selection-column.js

```diff
@@ -149,6 +162,132 @@ class GridSelectionColumn extends GridColumn {
     checkbox.indeterminate = this.__indeterminate;
   }
 
+  /** @private */
+  __lassoAutoScroller() {
+    if (this.__lassoDragStartIndex !== undefined) {
+      // Get the row being hovered over
+      const renderedRows = this._grid._getRenderedRows();
+      const hoveredRow = renderedRows.find((row) => {
+        const rowRect = row.getBoundingClientRect();
+        return this.__lassoCurrentY >= rowRect.top && this.__lassoCurrentY <= rowRect.bottom;
+      });
+
+      // Get the index of the row being hovered over or the first/last
+      // visible row if hovering outside the grid
+      let hoveredIndex = hoveredRow ? hoveredRow.index : undefined;
+      const scrollableArea = this.__getScrollableArea();
+      if (this.__lassoCurrentY < scrollableArea.top) {
+        hoveredIndex = this._grid._firstVisibleIndex;
+      } else if (this.__lassoCurrentY > scrollableArea.bottom) {
+        hoveredIndex = this._grid._lastVisibleIndex;
+      }
+
+      if (hoveredIndex !== undefined) {
+        // Select all items between the start and the current row
+        renderedRows.forEach((row) => {
+          if (
+            (hoveredIndex > this.__lassoDragStartIndex &&
+              row.index >= this.__lassoDragStartIndex &&
+              row.index <= hoveredIndex) ||
+            (hoveredIndex < this.__lassoDragStartIndex &&
+              row.index <= this.__lassoDragStartIndex &&
+              row.index >= hoveredIndex)
+          ) {
+            if (this.__lassoSelect) {
+              this._grid.selectItem(row._item);
+            } else {
+              this._grid.deselectItem(row._item);
+            }
+          }
```

Comment:
The current logic requires you to select two rows at minimum. Was this intended?  In V8 the first row gets immediately selected on mouse down. Should we at least select the first row as soon as the mouse is moved?

Reply:
IMO single row selection must be supported. Working on that

Reply:
Fixed